ssetests/tests_http_behavior.go

```diff
@@ -18,6 +23,26 @@ func DoHTTPBehaviorTests(t *ldtest.T) {
 			"Last-Event-Id header should not have had a value")
 	})
 
+	for _, status := range []int{301, 307} {
+		t.Run(fmt.Sprintf("client follows %d redirect", status), func(t *ldtest.T) {
+			server := NewStreamServer(t)
+
+			headers := make(http.Header)
+			headers.Set("Location", server.endpoint.BaseURL())
+			handler := httphelpers.HandlerWithResponse(status, headers, nil)
+			endpointReturningRedirect := requireContext(t).harness.NewMockEndpoint(handler, nil, t.DebugLogger())
```

Comment:
Here I'm using a capability that's provided by the underlying framework but wasn't previously being used in these tests: the ability to create any number of callback endpoints that aren't necessarily SSE streams. The test harness framework will take care of providing a unique URL; we only need to provide the request handler. `NewStreamServer` is using this under the covers.

Here I'm using a helper method that is equivalent to:

```go
handler := http.HandlerFunc(func(w http.ResponseWriter, r. *http.Request) {
    w.Header().Set("Location", server.endpoint.BaseURL())
    w.WriteHeader(status)
})
```